app/src/main/java/com/example/habitsmasher/DatePickerDialogFragment.java

```diff
@@ -0,0 +1,72 @@
+package com.example.habitsmasher;
+
+
+import android.app.DatePickerDialog;
+import android.app.Dialog;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Extraction of DatePickerDialog came from the following link:
+ *
+ * https://stackoverflow.com/a/34589270
+ * Post date: January 4, 2016
+ */
+
+public class DatePickerDialogFragment extends DialogFragment {
+
+    private static final String PATTERN = "dd/MM/yyyy";
+
+    private DatePickerDialog.OnDateSetListener _dateSetListener;
+
+    public DatePickerDialogFragment(DatePickerDialog.OnDateSetListener callback) {
+        _dateSetListener = callback;
+    }
+
+    @NonNull
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        DatePickerDialog datePickerDialog = new DatePickerDialog(
+                getActivity(),
+                this._dateSetListener,
+                Calendar.getInstance().get(Calendar.YEAR),
+                Calendar.getInstance().get(Calendar.MONTH),
+                Calendar.getInstance().get(Calendar.DAY_OF_MONTH)
```

Comment:
Minor: If we are calling `Calendar.getInstance()` three times here, we can just create a local variable so we don't have to fetch the instance multiple times, just once.
```
<type for calendar instance> calInstance = Calendar.getInstance();

// then, in the constructor for the DatePickerDialog you can use the variable you created
calInstance.get(Calendar.YEAR)
calInstance.get(Calendar.MONTH)
calInstance.get(Calendar.DAY_OF_MONTH)
```